### ISSUE TITLE
[335] Runeforging skill not rewarded properly, ranked 29987 / 0

### ISSUE THREAD

**Issue by MrSmite** [TrinityCore/TrinityCore#13882]
__Core:__ 42f1a0babfaacf64df31a934288979f8edfec940
__DB:__ TDB_full_335.56_2014_09_21.sql + updates

__Bug:__

When you create a DK and gain the [Runeforging](http://www.wowhead.com/spell=53428/runeforging) skill from completing quest [The Emblazoned Runeblade](http://www.wowhead.com/quest=12619), it starts you with a rank of __29987 / 0__ which results in your weapon displaying red text after applying a rune.

__Details:__

After using a runeforge, the weapon says the following in red:

> Rune of Razorice
> Modification requires runeforging

According to __.lookup skill__, runeforging is __776__ however this does not appear in *character_skills*. If you use the command __.setskill 776 255__ then it says "You have learned runeforging" and the red text turns green. It also adds the skill to the DB table.

NOTE: Once set, logging out / in, it sets the skill to 1/1 and moves it to the "Class skills" portion of the character sheet
NOTE: This worked properly in [previous versions](https://github.com/TrinityCore/TrinityCore/commit/c28345e279f31af2fa32df2411325ccfabe06fc8) of the core

**Comment by Exodius** [comment]
Runeforging is always 1/1.
If I am not mistaken, in 3.3.5a you gain Runeforging with few basic Runes already learned with some unlocking through levels that need to be learned through Class Trainer.

**Comment by MrSmite** [comment]
> Runeforging is always 1/1.

Right, once you use __setskill__ to give yourself the proper skill, the core resets it to 1/1 when you log out. The part that's broken is the quest not rewarding the skill properly.

**Comment by killradio** [comment]
@MrSmite Some time ago there was hack in core for lockpiking and runeforging to add this skills when character certain classes created. Ofc u can add values from skillLine dbcs, but for 434 core, for example, runeforging will add not only skill but all runes spells.

**Comment by tkrokli** [comment]
@MrSmite : the link "previous versions" in your OP is broken. The correct link is:
```
https://github.com/TrinityCore/TrinityCore/commit/c28345e279f31af2fa32df2411325ccfabe06fc8
```
You seem to have replaced the word 'commit' with 'issues', hence pointing to a 404 Page not found.
(btw, is that the HTML variant of a null pointer reference ?) ;)

**Comment by MrSmite** [comment]
@tkrokli 

I actually just pasted the hash into the OP and github generated the link automatically. Not sure why it picked the "issues". I fixed it and pasted the link you provided, thanks.

**Comment by tkrokli** [comment]
I just tested a bit to see if there would be any difference between learning
http://www.wowhead.com/spell=53431/runeforging (teaching spell) and
http://www.wowhead.com/spell=53428/runeforging (direct spell), I found that manually assigning 53431 adds 2 spell icons to the character's Spellbook while 53428 only adds 1. Still, it makes no difference regarding this issue (the red text and a random 5-digit skill level (out of 0) in the Runeforging UI).

The command that does what you want is: `.setskill 776 1` This makes the Runeforging UI show the character's Runeforging skill level as 1/1. No need to relog.

What we need to do now (I suppose), is to find where this specific quest action is handled and find out if it is possible to make that script/SAI/code run a command similar to `.setskill 776 1` in addition to the rest of the quest, so we can get this bug fixed. Does anyone have a pointer to where this is in the source?

**Comment by MrSmite** [comment]
According to the database it's a simple spell cast on quest completion:

quest `quest_template`.`id`=12619, `quest_template`.`RewardSpell`=53431, `quest_template`.`RewardSpellCast`=53431

**Comment by pete318** [comment]
After checking with a video https://youtu.be/ADiaJS585RM?t=135 and a discussion in IRC where Shauron located a sniff suggesting the same. The skill is learned (or at least set) when the quest is completed. My initial thinking was the skill should be added at character creation.

To that end, as also suggested in IRC, the AddSpell probably needs modifying just like for the rogue lockpicking skill:

https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Entities/Player/Player.cpp#L3815

I'll make a PR for the required change soon.

**Comment by tkrokli** [comment]
:+1: Nice PR, it looks quite promising. Hope to test it quite soon, just finish testing hunter Auto Shot first...

**Comment by MrSmite** [comment]
@pete318 - Nice PR, glad you noted that it isn't actually learned at char. creation (at least in 335 anyway).

**Comment by tkrokli** [comment]
Tested the PR on 3.3.5 and found it to be working as intended (tested once, going to repeat the test). :+1:

**Comment by mik1893** [comment]
Closed by #15116
